src/player/Player.php

```diff
@@ -2248,6 +2280,15 @@ public function getSaveData() : CompoundTag{
 			$nbt->setInt(self::TAG_SPAWN_Z, $spawn->getFloorZ());
 		}
 
+		if($this->deathPosition !== null){
+			$nbt->setString(self::TAG_DEATH_WORLD, $this->deathPosition->getWorld()->getFolderName());
```

Comment:
This may throw an exception if the world was unloaded since the player died

Reply:
Would you agree to store the world's name in a var on `Player::setDeathPosition()`?

Reply:
It should probably be done in a similar way to spawn position handling, where the spawn is invalidated if the world isn't loaded. I'm not sure this is the _best_ way, but it should be consistent.